File: ruby/testcode/loginjection_048.rb
require_relative 'shared'
require 'logger'

LOGGER = Logger.new($stdout)

# vuln-code-snippet start ruby_loginj_truncate_sanitize
def log_truncate_sanitize(req)
  sanitized = req.param('val').gsub(/[^\w\s@.\-]/, '')[0..100]
  LOGGER.info(sanitized) # vuln-code-snippet safe-line ruby_loginj_truncate_sanitize
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_loginj_truncate_sanitize

File: rust/testcode/loginjection_048.rs
//! CWE-117: Log helper discards user input and writes only a static message.

// vuln-code-snippet start testcodeLoginjection048
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let username = req.param("username");
    safe_log_event(&username); // vuln-code-snippet target-line testcodeLoginjection048
    super::shared::BenchmarkResponse::ok("OK")
}

fn safe_log_event(_user_input: &str) {
    eprintln!("[INFO] Login event received");
}
// vuln-code-snippet end testcodeLoginjection048
